### PR TITLE
feat: automatically publish on trunk branch

### PR DESCRIPTION
**Pull request by cdaringe**
# Problem

See #242

# Solution

- Break up CI files into a PR flow, & a `master`/trunk flow
- In the PR flow, do the exact same stuff as before, just don't run on changes to the master branch
- In the master flow, execute build & tests (as done in the PR flow), but additionally
    - auto publish
    - push incr version
    - skip ci for incremented version 

The following was auto-published using my fork on the `main` branch, executing the same publish flow: https://marketplace.visualstudio.com/items?itemName=cdaringe.vscode-jest-runner-dummy 

I personally opt for semantic release, but that's a pretty big personal behavioral change, so i stuck with the highly functional and basic approach for this round.

You can still increment the version for minors and majors as seen fit, CI will just always release with a patch rev on top of your minor or major versions.

### CODE REVIEW

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -24,4 +21,4 @@ jobs:
       - run: npm ci
       - run: npm run build --if-present
       - run: npm run test-compile
-      - run: npm run test
\ No newline at end of file
+      - run: npm run test
```
Comment (cdaringe) [inline .github/workflows/pull_request.yml:24]: `prettier` 🤦 😄

**Comment by firsttris** [comment]
ah i think you already prepared and it need to add it as VSCE_PAT correct?

**Comment by cdaringe** [comment]
Greetings. No worries. I’m probably not going to come back to this for minimally weeks.

ya, submission to marketplace is done, just needs the secret added in your settings as VSCE_PAT. See https://code.visualstudio.com/api/working-with-extensions/publishing-extension#get-a-personal-access-token

**Comment by firsttris** [comment]
hey @cdaringe 

i finally got it merged. 

there were some minor problems with tests, but i got it fixed :+1: 

sorry for taking so long, i was in the hospital for some weeks.

Do you want to become Collaborator in the Project? You would get full repo permissions. I need some help to take care of it.

best regards
Tristan

**Comment by cdaringe** [comment]
Hey @firsttris , ya im on board to help out

hope you’re recovering with hastily. godspeed!

**Pull request merged**